File: scripts/parity/scenarios/py/speed_changes.py
import fullcontrol as fc

# Test speed changes during print
steps = []
steps.append(fc.Point(x=30, y=30, z=0.2))
steps.append(fc.Printer(print_speed=1000))
steps.append(fc.Point(x=40, y=30, z=0.2))
steps.append(fc.Printer(print_speed=2000))
steps.append(fc.Point(x=50, y=30, z=0.2))
steps.append(fc.Printer(travel_speed=8000))
steps.append(fc.Extruder(on=False))
steps.append(fc.Point(x=60, y=30, z=0.2))
steps.append(fc.Extruder(on=True))
steps.append(fc.Point(x=70, y=30, z=0.2))

res = fc.transform(steps, result_type='gcode')
print(res)

File: scripts/parity/scenarios/js/speed_changes.mjs
// Parity scenario: speed changes during print
import { Point, Printer, Extruder, transform } from '../../../../dist/index.js'

// Test speed changes during print
const steps = []
steps.push(new Point({ x: 30, y: 30, z: 0.2 }))
steps.push(new Printer({ print_speed: 1000 }))
steps.push(new Point({ x: 40, y: 30, z: 0.2 }))
steps.push(new Printer({ print_speed: 2000 }))
steps.push(new Point({ x: 50, y: 30, z: 0.2 }))
steps.push(new Printer({ travel_speed: 8000 }))
steps.push(new Extruder({ on: false }))
steps.push(new Point({ x: 60, y: 30, z: 0.2 }))
steps.push(new Extruder({ on: true }))
steps.push(new Point({ x: 70, y: 30, z: 0.2 }))

const g = transform(steps, 'gcode', { show_tips: false, show_banner: false }).gcode
process.stdout.write(g.trimEnd() + '\n')
